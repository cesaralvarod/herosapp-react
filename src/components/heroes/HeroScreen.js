import React, { useMemo } from "react";
import { Redirect, useHistory, useParams } from "react-router-dom";
import getHerosById from "../../selectors/getHerosById";

const HeroScreen = () => {
  const { heroId } = useParams();

  const history = useHistory();

  const hero = useMemo(() => getHerosById(heroId), [heroId]);

  // const hero = getHerosById(heroId);

  if (!hero) return <Redirect to="/" />;

  const { superhero, publisher, alter_ego, first_appearance, characters } =
    hero;

  const handleReturn = () => {
    if (history.length < 2) {
      history.push("/");
    } else {
      history.goBack();
    }
  };

  return (
    <div className="container">
      <div className="row mt-5 mb-5">
        <div className="col-4 animate__animated animate__fadeInLeft">
          <img
            src={require(`../../assets/heroes/${heroId}.jpg`).default}
            alt={superhero}
            className="img-thumbnail"
          />
        </div>
        <div className="col-8">
          <h3>{superhero}</h3>
          <ul className="list-group list-group-flush">
            <li className="list-group-item">
              <b>Alter ego:</b>
              {alter_ego}
            </li>
            <li className="list-group-item">
              <b>Publisher:</b>
              {publisher}
            </li>
            <li className="list-group-item">
              <b>First appearence:</b>
              {first_appearance}
            </li>
          </ul>
          <h5>Characters: </h5>
          <p>{characters}</p>
          <button className="btn btn-outline-info" onClick={handleReturn}>
            Return
          </button>
        </div>
      </div>
    </div>
  );
};

export default HeroScreen;
