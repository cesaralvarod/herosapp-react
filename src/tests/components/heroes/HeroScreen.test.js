import React from "react";
import { mount } from "enzyme";
import HeroScreen from "../../../components/heroes/HeroScreen";
import { MemoryRouter, Route } from "react-router-dom";

describe("Testing in HeroScreen component", () => {
  test("Component should redirect to '/' if URL has not params", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/hero"]}>
        <HeroScreen />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find("Redirect").exists()).toBeTruthy();
  });

  test("Component should render an hero if URL has params", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/hero/marvel-spider"]}>
        <Route path="/hero/:heroId" component={HeroScreen} />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find(".row").exists()).toBeTruthy();
  });
});
