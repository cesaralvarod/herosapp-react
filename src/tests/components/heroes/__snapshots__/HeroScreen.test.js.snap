// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in HeroScreen component Component should redirect to '/' if URL has not params 1`] = `""`;

exports[`Testing in HeroScreen component Component should render an hero if URL has params 1`] = `
<div
  className="container"
>
  <div
    className="row mt-5 mb-5"
  >
    <div
      className="col-4 animate__animated animate__fadeInLeft"
    >
      <img
        alt="Spider Man"
        className="img-thumbnail"
      />
    </div>
    <div
      className="col-8"
    >
      <h3>
        Spider Man
      </h3>
      <ul
        className="list-group list-group-flush"
      >
        <li
          className="list-group-item"
        >
          <b>
            Alter ego:
          </b>
          Peter Parker
        </li>
        <li
          className="list-group-item"
        >
          <b>
            Publisher:
          </b>
          Marvel Comics
        </li>
        <li
          className="list-group-item"
        >
          <b>
            First appearence:
          </b>
          Amazing Fantasy #15
        </li>
      </ul>
      <h5>
        Characters: 
      </h5>
      <p>
        Peter Parker
      </p>
      <button
        className="btn btn-outline-info"
        onClick={[Function]}
      >
        Return
      </button>
    </div>
  </div>
</div>
`;
